### DISCUSSION TITLE
Cannot set `wire:dirty.class` and `wire:dirty.class.remove` on the same element

### DISCUSSION THREAD

**Discussion by samlev** [livewire/livewire#5420] [7. Bugs (archived)] [answered]
I'm trying to _replace_ a class on an element based on dirty state, and I've found that I can either add a new class _or_ remove the old class, but not both. As a minimal example:

```html
<input wire:model.defer="search">
<button
    class="text-gray-300"
    wire:click="$refresh"
    wire:target="search"
    wire:dirty.class="text-green-900"
    wire:dirty.class.remove="text-gray-300"
>
  Search
</button>
```

What I expect to happen is that `text-gray-300` is removed from the button, and `text-green-900` is added. What actually happens is that `text-green-900` is added, but `text-gray-300` also remains in place.

`wire:loading` acts as expected in this respect, and can also be used to remove dirty classes when it's loading. Here's an example with laravel playground that shows the different behaviours:

https://laravelplayground.com/#/snippets/b8bef312-681a-4a27-a2d1-628712edc3be

I think that this is because `DirtyStates` will only process the first `wire:dirty` attribute on an element and ignores any others: [DirtyStates set dirty state](https://github.com/livewire/livewire/blob/v2.10.7/js/component/DirtyStates.js#L63)

Unlike `wire:loading` which can handle multiple directives: [LoadingStates register hook](https://github.com/livewire/livewire/blob/v2.10.7/js/component/LoadingStates.js#L17-L23)

I'm not sure if there's a reason behind this (other than the "wip" commit message) or if this is something that you'd be happy to receive a PR over.

**Comment by joshhanley** [comment, marked as answer]
@samlev ah interesting! Yeah I'd expect it to work the same way as loading does. Yep PR's are welcome 🙂

**Comment by samlev** [comment]
PR: #5501
